chalice/awsclient.py

```diff
@@ -102,18 +133,45 @@ def _call_client_method_with_retries(self, method, kwargs):
         while True:
             try:
                 response = method(**kwargs)
-            except client.exceptions.InvalidParameterValueException:
+            except client.exceptions.InvalidParameterValueException as e:
                 # We're assuming that if we receive an
                 # InvalidParameterValueException, it's because
                 # the role we just created can't be used by
                 # Lambda so retry until it can be.
                 self._sleep(self.DELAY_TIME)
                 attempts += 1
-                if attempts >= self.LAMBDA_CREATE_ATTEMPTS:
+                message = e.response['Error'].get('Message', '')
+                # Do not retry if exceeded retries or does not have anything
```

Comment:
So a style nitpick, but I have said this several times before.  Rather than write a comment explaining *what* the code is doing, just put it in a separate function and omit the comment:

```
if self._is_iam_role_related_error(e):
    raise
```